pages/data/stations/_id.vue

```diff
@@ -213,9 +283,136 @@ export default {
     this.$store.dispatch('stations/downloadStations')
       .then(() => {
         this.populate()
+        // this.refreshMetrics()
       })
+    Promise.all([
+      this.$store.dispatch('stations/downloadStationsByDataset'),
+    ]).then(() => {
+      this.loadingStations = false
+      this.loadingMap = false
+      this.selectedYearRange = [
+        this.minSelectableYear, this.maxSelectableYear
+      ]
+      this.refreshMetrics()
+    })
   },
   methods: {
+    addToStartYear(amount) {
+      const newStartYear = this.selectedYearRange[0] + amount
+      this.setStartYear(newStartYear)
+    },
+    addToEndYear(amount) {
+      const newEndYear = this.selectedYearRange[1] + amount
+      this.setEndYear(newEndYear)
+    },
+    setEndYear(newEndYear) {
+      const oldStartYear = this.selectedYearRange[0]
+      let tempEndYear = this.maxSelectableYear
+
+      if (newEndYear >= this.maxSelectableYear) {
+        newEndYear = this.maxSelectableYear
+        tempEndYear = oldStartYear
+      } else if (newEndYear < oldStartYear) {
+        newEndYear = oldStartYear
+      }
+
+      // Set a temporary value, different from the current value, to force
+      // certain components to update.
+      this.selectedYearRange = [ oldStartYear, tempEndYear ]
+      this.$nextTick(() => {
+        // Set the actual intended value for the end year.
+        this.selectedYearRange = [ oldStartYear, newEndYear ]
+      })
+    },
+    setStartYear(newStartYear) {
+      const oldEndYear = this.selectedYearRange[1]
+      let tempStartYear = this.minSelectableYear
+
+      if (newStartYear <= this.minSelectableYear) {
+        newStartYear = this.minSelectableYear
+        tempStartYear = oldEndYear
+      } else if (newStartYear > oldEndYear) {
+        newStartYear = oldEndYear
+      }
+      // Set a temporary value, different from the current value, to force
+      // certain components to update.
+      this.selectedYearRange = [ tempStartYear, oldEndYear ]
+      this.$nextTick(() => {
+        // Set the actual intended value for the start year.
+        this.selectedYearRange = [ newStartYear, oldEndYear ]
+      })
+    },
+    stationText(station) {
+      const stationID = station.woudc_id || station.station_id
+      const stationName = station.name || station.station_name
+
+      if (this.stationOrder === 'woudc_id') {
+        return '(' + stationID + ') ' + stationName
+      } else { // name
+        return stationName + ' (' + stationID + ')'
+      }
+    },
+    async refreshMetrics() {
+      if (this.station === null) {
+        console.log('station is null')
```

Comment:
Remove `console.log()`; only to be used for dev purposes. Or replace with `console.error()` to notify an error on the UI